Code Quality: Remove hotlinking feature flag code

timarney opened the pull request:
## Context

This feature has been enabled by default in v1.23.0. 

<!-- What do we want to achieve with this PR? Why did we write this code? -->

## Summary

Removes feature flags for:

`videoPosterHotlinking`
`linkIconHotlinking`
`posterHotlinking`

<!-- A brief description of what this PR does. -->

## Relevant Technical Choices

<!-- Please describe your changes. -->

## To-do

<!-- A list of things that need to be addressed in this PR or follow-up changes. -->

## User-facing changes

<!--
Please describe your changes.
Include before/after screenshots or a short video.
-->

## Testing Instructions

<!--
How can the changes in this PR be verified?
Please provide step-by-step instructions how to reproduce the issue, if applicable.
Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->

<!-- ignore-task-list-start -->
- [ ] This is a non-user-facing change and requires no QA
<!-- ignore-task-list-end -->

This PR can be tested by following these steps:

1. Verify that hotlinking for story poster, video poster, and link icon is available and enabled and works for both contributors and admins
2. Other functionality should be left unchanged.

## Reviews

### Does this PR have a security-related impact?

No

<!-- Examples: new APIs, changes to KSES, etc.  -->

### Does this PR change what data or activity we track or use?

No

<!-- Examples: changes to telemetry, new third-party APIs -->

### Does this PR have a legal-related impact?

No

<!-- Examples: new images with unknown sources, new production dependencies with incompatible licenses -->

## Checklist

<!-- Check these after PR creation -->

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

<!--
Please reference the issue(s) this PR addresses.
No URLs, just the issue numbers.
Use "Fixes #123" if it fixes an issue.

NOTE: One reference per line!

Example:

Fixes #123
Partially addresses #456
See #789
-->

Fixes #11997

timarney committed:
remove hotlinking feature flag

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

swissspidy committed:
Trigger CI run now that the PR is ready for review

swissspidy commented on packages/wp-story-editor/src/components/documentPane/publish/publish.js:
```diff
@@ -366,7 +363,7 @@ function PublishPanel({ nameOverride }) {
                 onChangeErrorText={posterErrorMessage}
                 imgProps={featuredMedia}
-                canUpload={hasUploadMediaAction || enablePosterHotlinking}
+                canUpload={hasUploadMediaAction}
```
Isn't that a change in semantics?

Previously: `canUpload={hasUploadMediaAction || enablePosterHotlinking}`

But now we allow poster hotlinking by default (basically making `enablePosterHotlinking` always `true`), so technically this should be `canUpload={true}`, no?

timarney replied:
Good point -- updated.

swissspidy commented on packages/story-editor/src/components/publishModal/content/storyPreview.js:
```diff
@@ -319,7 +316,7 @@ const StoryPreview = () => {
                     onChangeErrorText={posterErrorMessage}
                     imgProps={featuredMedia}
-                    canUpload={hasUploadMediaAction || enablePosterHotlinking}
+                    canUpload={hasUploadMediaAction}
```
Same here, shouldn't this be `canUpload={true}`?

swissspidy commented on packages/story-editor/src/components/publishModal/content/storyPreview.js:
```diff
@@ -295,7 +292,7 @@ const StoryPreview = () => {
                   />
                 )}
-                {(hasUploadMediaAction || enablePosterHotlinking) && (
+                {hasUploadMediaAction && (
```
Same here, with poster hotlinking always allowed we can remove this entire condition here, no?

miina replied:
Not sure if you discussed this somewhere else that the comment was marked resolved, but since now `enablePosterHotlinking` is always `true`, wouldn't we always display this and can remove `hasUploadMediaAction` check as well?

timarney committed:
fix conditional

timarney committed:
check test

timarney committed:
revert after check

timarney committed:
check if MediaInput is valid

timarney committed:
remove check

timarney committed:
add condition

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

miina commented on includes/Experiments.php:
```diff
@@ -289,42 +289,6 @@ public function get_experiments(): array {
-			 * Creation date: 2022-06-24
-			 */
-			[
-				'name'        => 'linkIconHotlinking',
```
I can still see `videoPosterHotlinking` and `linkIconHotlinking` appearing in some Karma tests when doing a search in the whole codebase.

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

timarney committed:
canUpload prop

timarney committed:
re-add hasUploadMediaAction

timarney committed:
remove flag from karma

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

swissspidy committed:
Remove unnecessary `hasUploadMediaAction` check

Otherwise contributors without upload permissions cannot update the story poster via hotlinking.

Updates tests.

swissspidy committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag

timarney committed:
add poster var

timarney committed:
update publish test with media button

timarney committed:
Merge branch 'main' into fix/11997-remove-hotlinking-feature-flag
